test/spec/modules/33acrossBidAdapter_spec.js

```diff
@@ -1240,6 +1267,108 @@ describe('33acrossBidAdapter:', function () {
         expect(JSON.parse(builtServerRequests[0].data)).to.deep.equal(ttxRequest);
       });
     });
+
+    context('when user ID data exists as userIdAsEids Array in bidRequest', function() {
+      it('passes userIds in eids field in ORTB request', function() {
+        const eids = [
+          {
+            'source': 'x-device-vendor-x.com',
+            'uids': [
+              {
+                'id': 'yyy',
+                'atype': 1
+              },
+              {
+                'id': 'zzz',
+                'atype': 1
+              },
+              {
+                'id': 'DB700403-9A24-4A4B-A8D5-8A0B4BE777D2',
+                'atype': 2
+              }
+            ],
+            'ext': {
+              'foo': 'bar'
+            }
+          }
+        ];
+
+        const bidRequests = (
+          new BidRequestsBuilder()
+            .withUserIds(eids)
+            .build()
+        );
+
+        const ttxRequest = new TtxRequestBuilder()
+          .withUserIds(eids)
+          .withProduct()
+          .build();
+
+        const builtServerRequests = spec.buildRequests(bidRequests, {});
+
+        expect(JSON.parse(builtServerRequests[0].data)).to.deep.equal(ttxRequest);
+      });
+      it('does not validate eids ORTB', function() {
+        const eids = [1, 2, 3];
+
+        const bidRequests = (
+          new BidRequestsBuilder()
+            .withUserIds(eids)
+            .build()
+        );
+
+        const ttxRequest = new TtxRequestBuilder()
+          .withUserIds(eids)
+          .withProduct()
+          .build();
+
+        const builtServerRequests = spec.buildRequests(bidRequests, {});
+
+        expect(JSON.parse(builtServerRequests[0].data)).to.deep.equal(ttxRequest);
+      });
+    });
+
+    context('when user IDs do not exist under the userIdAsEids field in bidRequest as an Array', function() {
+      it('does not pass user IDs in the bidRequest ORTB', function() {
+        const eids = 'foo';
+
+        const bidRequests = (
+          new BidRequestsBuilder()
+            .withUserIds(eids)
+            .build()
+        );
+        bidRequests.userId = {
```

Comment:
Does `userId` here correspond to `user.id` in the request to TTX or is it just a dummy field to demonstrate that eids are not passed from another field?

Reply:
It's a real field that prebid core generates to pass 3P user ID. The user IDs are set in two places https://github.com/prebid/Prebid.js/issues/4166
The above setup asserts that our adapter only reads the `userIdAsEids` and not the submodule specific `userId.x`.